Source/DiabloUI/selgame.cpp

```diff
@@ -620,8 +623,8 @@ void selgame_Password_Select(int /*value*/)
 		} else {
 			strcpy(sgOptions.Network.szPreviousHost, selgame_Ip);
 		}
-		if (SNetJoinGame(selgame_Ip, gamePassword, gdwPlayerId)) {
-			if (!IsGameCompatibleWithErrorMessage(*m_game_data)) {
+		if (allowJoin && SNetJoinGame(selgame_Ip, gamePassword, gdwPlayerId)) {
+			if (!IsGameCompatibleWithErrorMessage(Gamelist[selgame_selectedGame - 3].gameData)) {

```

Comment:
This change only works for public games. We can use `m_game_data` here because it will be populated by `SNetJoinGame()` through a convoluted series of events starting with the `PT_JOIN_ACCEPT` handler and ending with the `EVENT_TYPE_PLAYER_CREATE_GAME` handler.

```suggestion
			if (!IsGameCompatibleWithErrorMessage(*m_game_data)) {
```